### ISSUE TITLE
Runing nodegit in Electron

### ISSUE THREAD

**Issue by Githubber2021** [nodegit/nodegit#1783]
Hi, I installed `nodegit` via `npm` and import the package in the renderer part of my Electron application. During compilation I receive this error below:

```
WARNING in ./node_modules/nodegit/dist/nodegit.js
Module not found: Error: Can't resolve '../build/Release/nodegit.node' in '/Users/steve/Documents/git/git_reader/node_modules/nodegit/dist'

ERROR in ./node_modules/nodegit/dist/nodegit.js
Module not found: Error: Can't resolve '../build/Debug/nodegit.node' in '/Users/steve/Documents/git/git_reader/node_modules/nodegit/dist'
ERROR in ./node_modules/nodegit/dist/nodegit.js
Module not found: Error: Can't resolve '../package' in '/Users/steve/Documents/git/git_reader/node_modules/nodegit/dist'
```

I found a variety of issues on the web regarding this, but I don't understand the solution. Is this caused by the Release build although I run a debug build of my electron app? Thank you very much!

**Comment by oyed** [comment]
Also seem to have this issue starting today.

**Comment by Githubber2021** [comment]
I created a repo, which I forked from a boilerplate template. I only added `nodegit` and `@types/nodegit` as a dependency and imported it in `details.component.ts`

https://github.com/Githubber2021/electron-nodegit-error

```
git clone https://github.com/Githubber2021/electron-nodegit-error.git
npm install
npm run electron:local
```

**Comment by seb-mtl** [comment]
I have the same issue, but on Windows. The error messages look the same

```
WARNING in ./node_modules/nodegit/build/Release/nodegit.node 1:2
Module parse failed: Unexpected character '�' (1:2)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
(Source code omitted for this binary file)

ERROR in ./node_modules/nodegit/dist/nodegit.js
Module not found: Error: Can't resolve '../build/Debug/nodegit.node' in 'D:\git\electron-nodegit-error\node_modules\nodegit\dist'
ERROR in ./node_modules/nodegit/dist/nodegit.js
```

**Comment by vIceBerg** [comment]
Hi.

Check if you have a reference to the package inside any page that is handled by the renderer. You cannot import nodegit in those source files, even if you're only using the types.

**Comment by oyed** [comment]
> Hi.
> 
> Check if you have a reference to the package inside any page that is handled by the renderer. You cannot import nodegit in those source files, even if you're only using the types.

I only import it inside the preload (Node enabled)

**Comment by Githubber2021** [comment]
Can you elaborate that in simpler terms? Does that mean I cannot use nodegit in the renderer part of my app? Just wondering, eg how GitLraken for instance does this

**Comment by vIceBerg** [comment]
> Thank you! Can you elaborate that in simpler terms? Does that mean I cannot use nodegit in the renderer part of my app? Just wondering, eg how GitLraken for instance does this

All calls and references must be made from the main process. So your renderer process should communicate with the main process somehow.

There's plenty of documentation online about this particular subject

**Comment by vIceBerg** [comment]
Can you try to install the 0.26.4 version and sees if it`s working? And are you sure the postinstall process calls electron-builder to rebuild dependencies to use with Electron?

**Comment by Githubber2021** [comment]
@vlceberg thank you for the explanation! I am just wondering, oyed said “suddenly” it doesn’t work for him anymore @oyed  did we just run into the same error message because my mistake, because it indeed seems I just made the simple mistake to execute that in the rendered part.

P.s. would the version downgrade have any difference to me then?

**Comment by vIceBerg** [comment]
> @vlceberg thank you for the explanation! I am just wondering, oyed said “suddenly” it doesn’t work for they anymore @oyed did we just run into the same error message because in my case, it seems I just made the simple mistake to import that in the render part.
> 
> P.s. would the version downgrade have any difference to me then?

I don't know. I use Electron and version 0.26.4 of nodegit and it works fine.

**Comment by vegarringdal** [comment]
I dont know if this helps.
But version 0.26.4 and 0.27.0 works fine as long as I run in in the background process/main.
If I try and run it in any frontend preload where I use nodejs express/fs etc it fails (dev console disconnects)

**Comment by yilikun** [comment]
> I dont know if this helps.
> But version 0.26.4 and 0.27.0 works fine as long as I run in in the background process/main.
> If I try and run it in any frontend preload where I use nodejs express/fs etc it fails (dev console disconnects)
I have the same problem.
And when I test it in a test repo,it works fine.
but when i import it in my electron project,it was broken.
Maybe this reason , but my preload will mount all modules....
